### ISSUE TITLE
mem Usage about CgroupMemStat 

### ISSUE THREAD

**Issue by upccup** [shirou/gopsutil#107]
hi @shirou  Very glad to see you add a metric related to docker. But there has a little question. Why do not you return docker memory like docker API 
```
"memory_stats" : {
        "stats" : {
           "total_pgmajfault" : 0,
           "cache" : 0,
           "mapped_file" : 0,
           "total_inactive_file" : 0,
           "pgpgout" : 414,
           "rss" : 6537216,
           "total_mapped_file" : 0,
           "writeback" : 0,
           "unevictable" : 0,
           "pgpgin" : 477,
           "total_unevictable" : 0,
           "pgmajfault" : 0,
           "total_rss" : 6537216,
           "total_rss_huge" : 6291456,
           "total_writeback" : 0,
           "total_inactive_anon" : 0,
           "rss_huge" : 6291456,
           "hierarchical_memory_limit" : 67108864,
           "total_pgfault" : 964,
           "total_active_file" : 0,
           "active_anon" : 6537216,
           "total_active_anon" : 6537216,
           "total_pgpgout" : 414,
           "total_cache" : 0,
           "inactive_anon" : 0,
           "active_file" : 0,
           "pgfault" : 964,
           "inactive_file" : 0,
           "total_pgpgin" : 477
        },
        "max_usage" : 6651904,
        "usage" : 6537216,
        "failcnt" : 0,
        "limit" : 67108864
     }
``` 
there has the max_usage and usage and limit clear.

**Comment by shirou** [comment]
Even the package name is `docker`, it does not limited to Docker, can get such as systemd or rocket. I aim to implement more general way, so currently I do not have an plan to use Docker specific API. (yah,  currently. your comments are always welcome!)

**Comment by upccup** [comment]
hi @shirou  Thanks for your reply.  But you may have misunderstood what I meant  I do not mean for you to call the docker's API. You can get docker memory usage in the file */sys/fs/cgroup/memory/docker/containerId/memory.limit_in_bytes*. I just mean in memory You can provide more info like * max_usage*, * usage*, * limit*, * failcnt*.

**Comment by shirou** [comment]
Oh, sorry, I got it now. Thank you! Actually, I was lazy to read multiple files :P 

It seems plenty of files under `/sys/fs/cgroup/memory/docker/containerId/`

- memory.failcnt
- memory.limit_in_bytes
- memory.max_usage_in_bytes
- memory.usage_in_bytes
- memory.kmem.failcnt
- memory.kmem.limit_in_bytes
- memory.kmem.max_usage_in_bytes
- memory.kmem.usage_in_bytes
- memory.kmem.tcp.failcnt
- memory.kmem.tcp.limit_in_bytes
- memory.kmem.tcp.max_usage_in_bytes
- memory.kmem.tcp.usage_in_bytes

Could you tell me some pointer which metrics should be include in the CgroupMemStat?

**Comment by upccup** [comment]
hi @shirou  I am glad to see your reply. I think add * memory.usage_in_bytes*, * memory.max_usage_in_bytes*, *memory.limit_in_bytes*, *memory.failcnt* will suffice. And now i am do some docker monitor. I think this info is important. Thank you.

**Comment by shirou** [comment]
Thank you. I will add these metrics until next weekend (hopefully).

**Comment by upccup** [comment]
@shirou Thank you very much.Is i need to  close this issue?

**Comment by shirou** [comment]
Nope. After implemented, I will close.

**Comment by shirou** [comment]
@upccup I have submitted PR #110 . Could you check it?

**Comment by anarcher** [comment]
:+1: Looks Good for me.

**Comment by upccup** [comment]
@shirou  :+1:

**Comment by shirou** [comment]
closed by #110. Thank you for the suggestion.
